benchmark/syclblas/blas2/trsv.cpp

```diff
@@ -0,0 +1,166 @@
+  const char* t_str = t.c_str();
+  const char* diag_str = diag.c_str();
+
+  index_t xlen = n;
+  index_t lda = n;
+  index_t incX = 1;
+
+  // The counters are double. We convert n to double to avoid
+  // integer overflows for n_fl_ops and bytes_processed
+  double n_d = static_cast<double>(n);
+
+  state.counters["n"] = n_d;
+
+  // Compute the number of A non-zero elements.
+  const double A_validVal = .5 * n_d * (n_d + 1);
+
+  {
+    double nflops = n_d * n_d;
+    state.counters["n_fl_ops"] = nflops;
+  }
+
+  {
+    double mem_readA = A_validVal;
+    double mem_readX = A_validVal;
+    double mem_writeX = A_validVal;
+    state.counters["bytes_processed"] =
+        (mem_readA + mem_readX + mem_writeX) * sizeof(scalar_t);
+  }
+
+  blas::SB_Handle& sb_handle = *sb_handle_ptr;
+
+  // Input matrix/vector, output vector.
+  std::vector<scalar_t> m_a(lda * n);
+  std::vector<scalar_t> v_x =
+      blas_benchmark::utils::random_data<scalar_t>(xlen);
+
+  // Populate the main diagonal with larger values.
+  for (int i = 0; i < n; ++i)
+    for (int j = 0; j < lda; ++j)
+      m_a[(i * lda) + j] = (i == j) ? blas_benchmark::utils::random_scalar(
+                                          scalar_t{9}, scalar_t{11})
+                                    : blas_benchmark::utils::random_scalar(
+                                          scalar_t{-10}, scalar_t{10}) /
+                                          scalar_t(n);
+
+  auto m_a_gpu = blas::make_sycl_iterator_buffer<scalar_t>(m_a, lda * n);
+  auto v_x_gpu = blas::make_sycl_iterator_buffer<scalar_t>(v_x, xlen);
+
+#ifdef BLAS_VERIFY_BENCHMARK
+  // Run a first time with a verification of the results
+  std::vector<scalar_t> v_x_ref = v_x;
+  reference_blas::trsv(uplo_str, t_str, diag_str, n, m_a.data(), lda,
+                       v_x_ref.data(), incX);
+  std::vector<scalar_t> v_x_temp = v_x;
+  {
+    auto v_x_temp_gpu =
+        blas::make_sycl_iterator_buffer<scalar_t>(v_x_temp, xlen);
+    auto event = _trsv(sb_handle, *uplo_str, *t_str, *diag_str, n, m_a_gpu, lda,
+                       v_x_temp_gpu, incX);
+    sb_handle.wait();
+  }
+
+  std::ostringstream err_stream;
+  if (!utils::compare_vectors(v_x_temp, v_x_ref, err_stream, "")) {
+    const std::string& err_str = err_stream.str();
+    state.SkipWithError(err_str.c_str());
+    *success = false;
+  };
+#endif
+
+  auto blas_method_def = [&]() -> std::vector<cl::sycl::event> {
+    auto event = _trsv(sb_handle, *uplo_str, *t_str, *diag_str, n, m_a_gpu, lda,
+                       v_x_gpu, incX);
+    sb_handle.wait(event);
+    return event;
+  };
+
+  // Warmup
+  blas_benchmark::utils::warmup(blas_method_def);
+  sb_handle.wait();
+
+  blas_benchmark::utils::init_counters(state);
+
+  // Measure
+  for (auto _ : state) {
+    // Run
+    std::tuple<double, double> times =
+        blas_benchmark::utils::timef(blas_method_def);
+
+    // Report
+    blas_benchmark::utils::update_counters(state, times);
+  }
+
```

Comment:
Thank you for your work @pgorlani 
Could you add  here and in clblast benchmark `state.SetItemsProcessed(state.iterations() * total_theoretical_flops);` and `state.SetBytesProcessed(state.iterations() * tot_mem_processed);`
 ?

Reply:
Thanks for your review, @s-Nick!

[f18ebd3](https://github.com/codeplaysoftware/sycl-blas/pull/381/commits/f18ebd3dbbbeee314b3ba11b3cb11fd54d799ca7) adds them.